python/lsst/daf/butler/sql/multipleDatasetQueryBuilder.py

```diff
@@ -97,9 +97,154 @@ class DatasetNecessityEnum(Enum):
     PREREQUISITE = auto()
 
 
-_SubqueryData = namedtuple("_SubqueryData", ("subquery", "necessity", "links"))
+class _SubqueryData(metaclass=ABCMeta):
+    """Helper interface for single-dataset subqueries or deferred follow-up
+    queries in `MultipleDatasetQueryBuilder`.
 
-_DeferralData = namedtuple("_DeferralData", ("builder", "necessity", "joinLinks", "perDatasetTypeLinks"))
+    Parameters
+    ----------
+    necessity : `DatasetNecessityEnum`
+        How the presence or absence of this dataset constrains the query.
+    links : `Set` of `str`
```

Comment:
`Set` -> `set`?

Reply:
I meant `collections.abc.Set`, but should spell that out.